### PR TITLE
64 login implement

### LINKED ISSUE #64: Implement user logging in on frontend side
This task depends on #30 #62 #63 
Should also update store state.

### PR DESCRIPTION
**Pull request by karolkozlo**
How to test: Try to sign in on some created account
closes #64 #120

### CODE REVIEW

File: frontend/src/pages/LoginPage.vue [hunk 1]
```diff
@@ -21,15 +21,22 @@
         </wof-button>
       </div>
     </div>
+    <div v-if="loading" class="loading">
+        <wof-spinner-dots :size="4" :loading="loading"></wof-spinner-dots>
+      </div>
```
Comment (kpagacz) [inline frontend/src/pages/LoginPage.vue:26]: THE  DOTS XD

File: frontend/src/pages/LoginPage.vue [hunk 2]
```diff
@@ -57,12 +82,24 @@ export default {
...
-        //This will be replaced with call to http
-        console.log("Username:"+this.username.value);
-        console.log("Password:"+this.password.value);
+        try {
+          await this.$store.dispatch('logIn', {username: this.username.value, password: this.password.value});
+          this.$router.push(`/user/${this.username.value}`);
+        } catch(err) {
+          this.resultTitle = 'Error';
+          this.resultType = 'error';
+          this.resultMessage = err.message;
```
Comment (kpagacz) [inline frontend/src/pages/LoginPage.vue:94]: This is going to make much more sense (the error message) once we deal with sending sensible error messages from the backend, right?

Comment (kpagacz) [reply frontend/src/pages/LoginPage.vue:94]: Because currently it's "Error logging in"

Comment (karolkozlo) [reply frontend/src/pages/LoginPage.vue:94]: This error is mainly for network errors or something like that. Errors for invalid passwords or usernames will be displayed on form inputs. So yes, this is going to make more sense

**Review by kpagacz — CHANGES_REQUESTED** [review]
Splendid job! This looks very good!

Would you mind adding some documentation to this new component, store mutations, actions etc? Would be of much help, especially if you included some examples of usage (could be just a copy-paste from the login component you have created).

Some ideas:
* make the login button be pressed when user clicks enter and password input is active (so a user doesn't have to use the mouse to click on login or tab to the button)
* the dots spinner is shown very quickly and disappears similarly quickly. It'a a blink of an eye. It's quite jarring to be honest and leaves me no time to parse the dots even, so it leaves me a little confused. Some ideas to resolve it: don't hide the whole login input, just disable the login button and replace it/overlay dots on top of the button only, leaving the rest of the component still shown (with possibly disabled inputs). Option number two would be to add a transition to showing/hiding - could ease-in changes in opacity impose a minimum time that the dots should be shown? 

And the sign out button works as well, so it seems we are set until the new backend endpoints are available.

**Comment by karolkozlo** [comment]
> Option number two would be to add a transition to showing/hiding - could ease-in changes in opacity impose a minimum time that the dots should be shown?

Yeah, I will add a minimum amount of time during which the dots will be visible

**Review by kpagacz — CHANGES_REQUESTED** [review]
Would you mind taking a look at this inline documentation framework and if possible adhering to their style? https://vue-styleguidist.github.io/docs/Documenting.html#code-comments

File: giddy-banana/config/config.json [hunk 3]
```diff
@@ -5,8 +5,7 @@
     "database": "db",
     "host": "localhost",
     "dialect": "mysql",
-    "port": "3306",
-    "seederStorage": "sequelize"
+    "port": "3306"
```
Comment (kpagacz) [inline giddy-banana/config/config.json:8]: That should stay as is on the remote

File: frontend/src/pages/LoginPage.vue [hunk 4]
```diff
@@ -40,15 +46,42 @@ export default {
         value: '',
         errorMsg: ''
       },
+      loading: false,
+      /**
+        * All those result{X} properiets are related to error on logging in
```
Comment (kpagacz) [inline frontend/src/pages/LoginPage.vue:51]:
```suggestion
        * All those result{X} properties are related to error on logging in
```

**Review by kpagacz — CHANGES_REQUESTED** [review]
Looks better with the docs!

One small typo in the docs and one big thing in the backend - the giddy files should stay as they are on the remote otherwise thir PR will break `npm run dev`.

Otherwise, looks good. I will approve, once you walk back that change in giddy

**Pull request merged**